test/alt.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
```

Comment:
It strikes me that it would be nice to test these typeclass functions on as minimal a case as possible i.e. the `FreeFoo`.

Certainly out of scope for this PR, and I'm not sure where you get your functors from, but what do you think about it in principle?